agentendpoint/inventory.go

```diff
@@ -61,23 +62,28 @@ func (c *Client) report(ctx context.Context, state *inventory.InstanceInventory)
 	inventory := formatInventory(ctx, state)
 
 	reportFull := false
-	retries := 0
+	attempts := 0
+	var res *agentendpointpb.ReportInventoryResponse
+	var err error
+	// Retry each reportInventory call, and report full inventory no more than maxRetries times.
 	for {
-		resp, err := c.reportInventory(ctx, inventory, reportFull)
-		if err != nil {
+		if err = retryutil.RetryAPICall(ctx, apiRetrySec*time.Second, "ReportInventory", func() error {
+			res, err = c.reportInventory(ctx, inventory, reportFull)
+			if err != nil {
+				return err
+			}
+			clog.Debugf(ctx, "ReportInventory response:\n%s", util.PrettyFmt(res))
+			return nil
+		}); err != nil {
 			clog.Errorf(ctx, "Error reporting inventory: %v", err)
+			break
+		} else if attempts > maxRetries {
```

Comment:
I am curious, why do we have a logic of reporting inventory multiple times. Shouldn't the logic be 
1. Call ReportInventory with reportFull = false
2. If the ReportFullInventory is true, call ReportInventory with reportFull = true.

In what case would the agent be asked to report inventory again? Wouldn't that mean a bug in the service?

Reply:
Each response comes back with a reportFullInventory flag which might still be `true` even if full inventory was just sent, most likely due to unforeseen issues in either agent not sending or the service not receiving inventory as expected.

We could disregard the flag in the second request and try again next time inventory task runs on the agent if we are ok with that behavior.

Reply:
I would suggest to change the behavior with what I suggested in the earlier comment instead of blindly retrying. We dont know how retries would solve the unforeseen issues in either the agent or service except from rpc retrying at the request level for the different codes we receive from the service, which we have in place.

Reply:
Simplified!